packages/state/src/utils/index.ts

```
@@ -2,17 +2,27 @@ import { Type as RecsType, Schema } from "@dojoengine/recs";
+        let schemaType = schema[key];
+        let value = values[key];
 
         if (value == null) {
             acc[key] = value;
             return acc;
         }
 
         if (value.type === "enum") {
-            acc[key] = value.value.option;
-            return acc;
+            if (value.type_name && value.type_name.includes("Option")) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)